### PR TITLE
Use XFCE for STATA Desktop and add `install.do` support

### LINKED ISSUE #102: Change XPRA environments to open Desktop
For discussion

* Stata does not natively allow for a terminal to be opened (the way that Rstudio does). Could the XPRA environment be configured to at least allow for a terminal to be opened?  
* Working via the Stata command line is kludgy.
* Closing the Stata instance requires restarting the tale
* Just accidentally killed my Stata instance ... have to restart the whole Tale. 
* If not desktop, then XPRA needs a menu that allows to open (a) Stata (b) Terminal.

Today the XPRA environments start the app itself, but this could be changed to start a Linux Desktop instead:
https://github.com/whole-tale/terraform_deployment/blob/master/migration/migrate_v1.0.py#L242

### PR DESCRIPTION
**Pull request by craig-willis**
Fixes https://github.com/whole-tale/whole-tale/issues/102 and https://github.com/whole-tale/repo2docker_wholetale/issues/25

Depends on/related to:
* https://github.com/whole-tale/repo2docker/pull/4
* https://github.com/whole-tale/gwvolman/pull/154

This PR:
* Changes the STATA Xpra environment to use a full XFCE4 desktop
* Adds support for `install.do` for package installation at build time via `ssc install` or `net install`. These packages are installed to the SITE sysdir (i.e., shared/network install dir)
* Packages installed interactively will be ephemerally installed in `~`. (Note: Even if desirable, we can't support direct install to the workspace directory since STATA uses `link()` which is not supported by WebDav)

**To test locally**

Clone  https://github.com/whole-tale/repo2docker/pull/4 and `pip install -e .`

Clone this repo and `pip install -e .`

Encode STATA license to pass as build arg (this is done in https://github.com/whole-tale/gwvolman/pull/154):
```
STATA_LICENSE_ENCODED=`base64 -w 0 /home/ubuntu/deploy-dev/volumes/licenses/stata/stata.<date>.lic`
```

Build the image using the example config:
```
cd repo2docker_examples/stata
repo2docker --build-arg STATA_LICENSE_ENCODED=${STATA_LICENSE_ENCODED} --config \
/home/ubuntu/repo2docker_wholetale/repo2docker_config.py --user-name jovyan \
--user-id 1000 --debug --no-run .
```

You should see the following where the `latab` package is installed successfully to `/usr/local/stata/ado/site`
```
#48 1.756 Running /usr/local/stata/profile.do ...
#48 1.756
#48 1.756 . sysdir set PLUS /usr/local/stata/ado/site
#48 1.756
#48 1.756
#48 1.756 . ssc install latab
#48 2.420 checking latab consistency and verifying not already installed...
#48 3.121 installing into /usr/local/stata/ado/site/...
#48 3.122 installation complete.
#48 3.122
#48 3.122 . #48 DONE 4.2s
```

Run the image
```
docker run -p 10001:10000 -v /home/ubuntu/deploy-dev/volumes/licenses/stata/stata.<date>.lic:/usr/local/stata/stata.lic <image name>  xpra start-desktop --bind-tcp=0.0.0.0:10000 --html=on --daemon=no --exit-with-children=no --start=xfce4-session
```

Confirm `latab` is indeed installed
```
ls /usr/local/stata/ado/site/l
latab.ado  latab.hlp  latabstat.ado  latabstat.hlp
```

Install a package interactively:
```
stata do test.do

...
. ssc install outreg2
checking outreg2 consistency and verifying not already installed...
installing into /home/jovyan/ado/plus/...
installation complete.
...

ls ~/ado/plus/o/
```
Confirm outreg2 is installed in the PLUS sysdir.

**To test on Staging**
My gwvolman and repo2docker_wholetale images have been deployed to staging.

* Go to https://dashboard.stage.wholetale.org
* Create a new STATA 16 Desktop tale
* Create `install.do` file in workspace with line `ssc install latab`
* Rebuild tale. Inspect build logs to confirm that ssc install latab was successful to `/usr/local/stata/ado/site/`
* Run stata and `ssc install outreg2`. Confirm files are installed to `~/ado/plus`

### CODE REVIEW

File: repo2docker_wholetale/stata.py [hunk 1]
```diff
@@ -65,8 +67,104 @@ def get_build_scripts(self):
...
+                > ${HOME}/Desktop/Firefox.desktop && \
+                chmod +x ${HOME}/Desktop/*.desktop
+                """,
+            ),
+            (
+                "root",
+                r"""
+                mkdir -p /usr/local/stata/ado/plus && \
+                mkdir -p /usr/local/stata/ado/site && \
+                sleep 60 && \
```
Comment (Xarthisius) [inline repo2docker_wholetale/stata.py]: why sleep?

File: repo2docker_wholetale/stata.py [hunk 2]
```diff
@@ -65,8 +67,104 @@ def get_build_scripts(self):
...
+        """
+        scripts = []
+
+        installdo_path = self.binder_path("install.do")
+        if os.path.exists(installdo_path):
+            scripts += [
+                (
+                    "root",
+                    r"""
+                    ls /usr/local/stata && which stata &&\
```
Comment (Xarthisius) [inline repo2docker_wholetale/stata.py]: looks like debug statement to me, unless I'm missing something

**Review by Xarthisius — CHANGES_REQUESTED** [review]

File: repo2docker_wholetale/stata.py (hunk 1 above)
Comment (craig-willis) [reply repo2docker_wholetale/stata.py]: Ack. Trash attempt to debug a build issue.

File: repo2docker_wholetale/stata.py (hunk 2 above)
Comment (craig-willis) [reply repo2docker_wholetale/stata.py]: Indeed. Removed.

**Review by Xarthisius — APPROVED** [review]

**Pull request merged**